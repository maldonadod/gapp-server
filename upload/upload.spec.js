jest.mock('busboy')
jest.mock('cloudinary')
const middleware = require('./middleware')
const cloudinary = require('cloudinary')
const {
  chapterCoverUploadFormat
} = require('../src/Chapters/business')

jasmine.DEFAULT_TIMEOUT_INTERVAL = 5

function makeCallback(done, body) {
  return (...args) => {
    try {
      body(...args);
      done();
    } catch (error) {
      done.fail(error);
    }
  };
}

describe('Upload', () => {

  test('middleware', done => {
    expect.assertions(2)
    const chapter = {
      description: 'The description'
    }
    const req = {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
      ,pipe: thing => thing.feed({fieldname: 'cover', fields: chapter})
    }
    const res = {}
    const next = jest.fn(makeCallback(done, () => {
      const cover = cloudinary.getCover()
      expect(res.locals).toMatchSnapshot()
      expect(next.mock.calls.length).toEqual(1)
    }))

    middleware(chapterCoverUploadFormat)(req, res, next)
  })
})
