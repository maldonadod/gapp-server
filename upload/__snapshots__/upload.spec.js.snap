// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload middleware 1`] = `
Object {
  "cover": Object {
    "url": "http://localhost/",
  },
  "description": "The description",
}
`;
